src/modules/capabilities.json

```diff
@@ -0,0 +1,704 @@
+                            {
+                                "capability": "xrn:firebolt:capability:token:platform",
+                                "role": "use",
+                                "supported": true,
+                                "available": false,
+                                "permitted": false,
+                                "granted": false,
+                                "details": ["unavailable"]
+                            },
+                            {
+                                "capability": "xrn:firebolt:capability:protocol:moca",
+                                "role": "use",
+                                "supported": true,
+                                "available": false,
+                                "permitted": false,
+                                "granted": false,
+                                "details": ["disabled", "unavailable"]
+                            },
+                            {
+                                "capability": "xrn:firebolt:capability:wifi:scan",
+                                "role": "use",
+                                "supported": true,
+                                "available": true,
+                                "permitted": false,
+                                "granted": false,
+                                "details": ["unpermitted"]
+                            },
+                            {
+                                "capability": "xrn:firebolt:capability:localization:postalCode",
+                                "role": "use",
+                                "supported": true,
+                                "available": true,
+                                "permitted": true,
+                                "granted": false,
+                                "details": ["ungranted"]
+                            },
+                            {
+                                "capability": "xrn:firebolt:capability:localization:postalCode",
+                                "role": "use",
+                                "supported": true,
+                                "available": true,
+                                "permitted": true,
+                                "granted": false,
+                                "details": ["ungranted"]
+                            },
+                            {
+                                "capability": "xrn:firebolt:capability:localization:locality",
+                                "role": "use",
+                                "supported": true,
+                                "available": true,
+                                "permitted": true,
+                                "granted": false,
+                                "details": ["grantDenied", "ungranted"]
+                            }
+                        ]
+                    }
+                }
+            ]
+        },
+        {
+            "name": "request",
```

Comment:
so @jlacivita I think we discussed this in another PR. But this API doesn't really make sense for single-use grant lifetimes. This API is meant to prime the app's grants upfront so they can use them later. But you can't prime a single-use grant because it will already be revoked by the time you use the capability. Maybe we don't need to do anything for this, it just won't work for those types of grant life times? Was also wondering if this API should return a FAT so the app can actually use the single use grants when this returns.

Reply:
I was debating whether to update this API to return a FAT. I think there's two use cases:

- I'm an app that wants to request user grants at some point before calling the APIs gated by them
- I'm an extension SDK that wants a FAT token, with relevant grants as needed, to pass to a Firebolt cloud API.

For the former, I was thinking this method. For the latter, I was thinking we'd make a new method called `Capabilities.authorize` that takes a list of required and optional capabilities, get's any needed grants, and returns a token if all of the required capabilities were supported/available/permitted/granted. Optional capabilities would be included in the FAT if they were supported/available/permitted/granted. The method would error out if any required caps are not s/a/p/g.